lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -302,13 +304,13 @@ class DocumentImpl extends NodeImpl {
     return (this._parsingMode === "xml" || this.doctype) ? "CSS1Compat" : "BackCompat";
   }
   get charset() {
-    return "UTF-8";
+    return this._encoding || this._defaultEncoding;
```

Comment:
I think we should not have a _defaultEncoding option, but instead set _encoding to be either the passed encoding or the passed defaultEncoding.